### PR TITLE
Add collector status

### LINKED ISSUE #1573: Add Collector status to the collector page
This issue is a part of:

- #1574 

For prior and relevant discussion see, https://github.com/open-telemetry/opentelemetry.io/issues/1570#issuecomment-1199491412, and https://github.com/open-telemetry/opentelemetry.io/issues/1570#issuecomment-1199506433 in which @TylerHelmuth wrote:
> Makes sense. The collector SIG has established [Stability Levels](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) for components and is https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10116 with that status (and more). There has also been updates to allow components to print out their stability level on collector startup if the component is unmaintained or deprecated. Hopefully that work is helpful in some way.

### PR DESCRIPTION
**Pull request by chalin**
A proposal for the collector-status section.

- Closes #1573
- Contributes to #1574

Preview: https://deploy-preview-1576--opentelemetry.netlify.app/docs/collector/#status-and-releases

/cc @open-telemetry/collector-contrib-approvers @cartermp @TylerHelmuth @austinlparker

### CODE REVIEW

File: content/en/docs/collector/_index.md [hunk 1]
```diff
@@ -40,4 +39,16 @@ can take care of additional handling like retries, batching, encryption or even
 
 It is also easier to [setup a collector](./getting-started) than you might think: the default OTLP exporters in each language assume a local collector endpoint, so you'd start up a collector and you'd just start getting telemetry.
 
+## Status and releases
+
+The **collector** status is: [Stable][].
```
Comment (codeboten) [inline content/en/docs/collector/_index.md:44]: The collector has not yet reached stable status, as not all of its core components are stable.

Comment (chalin) [reply content/en/docs/collector/_index.md:44]: Thanks. What would be the correct status?

**Review by cartermp — COMMENTED** [review]
I think this looks good overall, but we'll want a label to use for the overall status from collector maintainers before we can proceed.

File: content/en/docs/collector/_index.md (hunk 1 above)
Comment (chalin) [reply content/en/docs/collector/_index.md:44]: Maybe the current status might be considered as [**mixed**](https://opentelemetry.io/docs/reference/specification/document-status/#mixed)?

Comment (chalin) [reply content/en/docs/collector/_index.md]: > The collector has not yet reached stable status, as not all of its core components are stable.

I've updated the status to "mixed", which is an official spec term :). PTAL.

**Review by cartermp — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Comment by chalin** [comment]
Thanks all for your feedback.

**Pull request merged**